File: src/URI/URI_1094.java
package URI;
import java.util.Scanner;

public class URI_1094 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int totCoelho = 0, totRato = 0, totSapo = 0;
		int n = sc.nextInt();
		
		for(int i = 0; i < n; i++) {
			int qtde = sc.nextInt();
			char animal = sc.next().charAt(0);
			if(animal == 'S') {
				totSapo += 1 * qtde;
			}
			else if(animal == 'R') {
				totRato += 1 * qtde;
			}
			else {
				totCoelho += 1 * qtde;
			}
		}
		int cobaias = totSapo + totRato + totCoelho;
		System.out.println("Total: " + cobaias + " cobaias");
		System.out.println("Total de coelhos: " + totCoelho);
		System.out.println("Total de ratos: " + totRato);
		System.out.println("Total de sapos: " + totSapo);
		
		double percentualCoelho = (double) totCoelho / cobaias * 100;
		System.out.printf("Percentual de coelhos: %.2f %%%n", percentualCoelho );
		double percentualRato = (double) totRato / cobaias * 100;
		System.out.printf("Percentual de ratos: %.2f %%%n", percentualRato );		
		double percentualSapo = (double) totSapo / cobaias * 100;
		System.out.printf("Percentual de sapos: %.2f %%%n", percentualSapo );		
		
		sc.close();

	}

}

File: kotlin/URI_1094.kt
import java.util.*

fun main(args: Array<String>) {
    Locale.setDefault(Locale.US)
    val sc = Scanner(System.`in`)
    val cobaias = mutableMapOf<String, Int>()
    val n = sc.nextInt()
    repeat(n) {
        val quantia = sc.nextInt()
        val tipo = sc.next()
        val quantiaDoTipo = cobaias[tipo] ?: 0
        cobaias[tipo] = quantia+quantiaDoTipo
    }

    val coelhos = cobaias["C"] ?: 0
    val ratos = cobaias["R"] ?: 0
    val sapos = cobaias["S"] ?: 0
    val total = coelhos + ratos + sapos
    val percentualCoelhos = coelhos / (total*1.0) * 100
    val percentualRatos = ratos / (total*1.0) * 100
    val percentualSapos = sapos / (total*1.0) * 100

    println("Total: $total cobaias")
    println("Total de coelhos: $coelhos")
    println("Total de ratos: $ratos")
    println("Total de sapos: $sapos")
    println("Percentual de coelhos: ${"%.2f".format(percentualCoelhos)} %")
    println("Percentual de ratos: ${"%.2f".format(percentualRatos)} %")
    println("Percentual de sapos: ${"%.2f".format(percentualSapos)} %")
}
